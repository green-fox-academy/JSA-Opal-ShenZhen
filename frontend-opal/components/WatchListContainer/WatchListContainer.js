import React from 'react';
import { View } from 'react-native';
import WatchList from '../WatchList/WatchList';
import Header from '../NewHeader/newHeader';
import NavBar from '../Header/NavBar';

export default function WatchListContainer() {
  const info = [{
    name: 'MSFT', num: '141.21', increase: '+0.23', whole: '1.2M',
  },
  {
    name: 'AMZN', num: '2108.2', increase: '-0.06', whole: '609K',
  },
  {
    name: 'AAPL', num: '250.7', increase: '+1.63', whole: '810K',
  }];

  const chartData = [
    { x: 1, y: 8 },
    { x: 2, y: 7 },
    { x: 3, y: 5 },
  ];

  const stockData = {
    low: '231.6',
    high: '250.7',
    Pe: '20.3',
    EPS: '1.2',
    DivYield: '2.34%',
  };

  const watchlistArr = ['my watchlist1', 'my watchlist2'];

  const navTitles = [
    { id: 1, key: 'PORTFOLIO' },
    { id: 2, key: 'WATCHLIST' },
  ];

  return (
    <View>

      <Header />
      <NavBar navTitles={navTitles} />
      <WatchList
        watchlistArr={watchlistArr}
        info={info}
        stockData={stockData}
        chartData={chartData}
      />
    </View>
  );
}
