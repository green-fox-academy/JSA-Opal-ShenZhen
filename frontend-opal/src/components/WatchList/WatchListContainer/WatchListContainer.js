import React from 'react';
import { View } from 'react-native';
import { connect } from 'react-redux';

import WatchList from '../WatchList/WatchList';

function WatchListContainer() {
  const watchlists = ['my watchlist1', 'my watchlist2'];

  return (
    <View>
      <WatchList watchlists={watchlists} />
    </View>
  );
}

WatchListContainer.title = 'WATCHLIST';

export default connect(null, null)(WatchListContainer);
