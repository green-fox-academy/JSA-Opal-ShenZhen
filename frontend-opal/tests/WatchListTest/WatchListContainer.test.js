import React from 'react';
import renderer from 'react-test-renderer';

import WatchListContainer from '../../src/components/WatchList/WatchListContainer/WatchListContainer';

describe('<WatchListContainer />', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<WatchListContainer />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
