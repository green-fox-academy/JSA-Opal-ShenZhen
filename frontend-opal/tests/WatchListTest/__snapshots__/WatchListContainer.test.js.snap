// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<WatchListContainer /> renders correctly 1`] = `
<View>
  <View
    onLayout={[Function]}
  >
    <View
      style={
        Object {
          "backgroundColor": "#F8F8F8",
          "borderBottomColor": "#a7a6ab",
          "borderBottomWidth": 0.5,
          "elevation": 3,
          "flexDirection": "row",
          "height": 64,
          "justifyContent": "center",
          "left": 0,
          "paddingLeft": 6,
          "paddingRight": 10,
          "paddingTop": 18,
          "right": 0,
          "shadowColor": undefined,
          "shadowOffset": undefined,
          "shadowOpacity": undefined,
          "shadowRadius": undefined,
          "top": 0,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "flex-start",
            "alignSelf": "center",
            "flex": 1,
          }
        }
      >
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "alignSelf": null,
              "backgroundColor": "transparent",
              "borderBottomWidth": null,
              "borderColor": null,
              "borderLeftWidth": null,
              "borderRadius": 5,
              "borderRightWidth": null,
              "borderTopWidth": null,
              "borderWidth": undefined,
              "elevation": 0,
              "flexDirection": "row",
              "height": 45,
              "justifyContent": "space-between",
              "marginLeft": -3,
              "opacity": 1,
              "paddingBottom": 6,
              "paddingLeft": 4,
              "paddingRight": 6,
              "paddingTop": 6,
              "shadowColor": null,
              "shadowOffset": null,
              "shadowOpacity": null,
              "shadowRadius": null,
            }
          }
        >
          <Text />
        </View>
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "alignSelf": "center",
            "flex": 1,
          }
        }
      >
        <Text
          numberOfLines={1}
          style={
            Object {
              "color": "#000",
              "fontFamily": "System",
              "fontSize": 17,
              "fontWeight": "700",
              "marginLeft": undefined,
              "paddingLeft": 4,
              "paddingTop": 1,
              "textAlign": "center",
            }
          }
        >
          Header
        </Text>
      </View>
      <View
        style={
          Object {
            "alignItems": "flex-end",
            "alignSelf": "center",
            "flex": 1,
            "flexDirection": "row",
            "justifyContent": "flex-end",
          }
        }
      >
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "alignSelf": null,
              "backgroundColor": "transparent",
              "borderBottomWidth": null,
              "borderColor": null,
              "borderLeftWidth": null,
              "borderRadius": 50,
              "borderRightWidth": null,
              "borderTopWidth": null,
              "borderWidth": undefined,
              "elevation": 0,
              "flexDirection": "row",
              "height": 45,
              "justifyContent": "space-between",
              "marginRight": -9,
              "opacity": 1,
              "paddingBottom": 6,
              "paddingLeft": 15,
              "paddingRight": 15,
              "paddingTop": 6,
              "shadowColor": null,
              "shadowOffset": null,
              "shadowOpacity": null,
              "shadowRadius": null,
            }
          }
        >
          <Text />
        </View>
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "display": "flex",
        "flexDirection": "row",
        "justifyContent": "space-between",
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "fontSize": 12,
            "height": 50,
            "padding": 2,
            "textAlign": "center",
            "width": "50%",
          },
          Object {
            "borderBottomColor": "#e17055",
            "borderBottomWidth": 1,
          },
        ]
      }
    >
      <View
        accessibilityRole="button"
        accessibilityStates={Array []}
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Array [
              Object {},
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#007AFF",
                  "fontSize": 18,
                  "padding": 8,
                  "textAlign": "center",
                },
                Object {
                  "color": "#e17055",
                },
              ]
            }
          >
            PORTFOLIO
          </Text>
        </View>
      </View>
    </View>
    <View
      style={
        Array [
          Object {
            "fontSize": 12,
            "height": 50,
            "padding": 2,
            "textAlign": "center",
            "width": "50%",
          },
          false,
        ]
      }
    >
      <View
        accessibilityRole="button"
        accessibilityStates={Array []}
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Array [
              Object {},
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#007AFF",
                  "fontSize": 18,
                  "padding": 8,
                  "textAlign": "center",
                },
                Object {
                  "color": "#e17055",
                },
              ]
            }
          >
            WATCHLIST
          </Text>
        </View>
      </View>
    </View>
  </View>
  <View>
    <View>
      <View>
        <View
          style={
            Object {
              "borderColor": "black",
              "borderStyle": "solid",
              "borderWidth": 2,
              "display": "flex",
              "flexDirection": "row",
              "justifyContent": "space-between",
            }
          }
        >
          <Text>
            my watchlist1
          </Text>
          <View
            accessible={true}
            isTVSelectable={true}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#007aff",
                "borderBottomWidth": null,
                "borderColor": "#007aff",
                "borderLeftWidth": null,
                "borderRadius": 5,
                "borderRightWidth": null,
                "borderTopWidth": null,
                "borderWidth": undefined,
                "elevation": 2,
                "flexDirection": "row",
                "height": 45,
                "justifyContent": "space-between",
                "opacity": 1,
                "paddingBottom": 6,
                "paddingTop": 6,
                "shadowColor": undefined,
                "shadowOffset": undefined,
                "shadowOpacity": undefined,
                "shadowRadius": undefined,
              }
            }
          >
            <Text />
          </View>
        </View>
      </View>
    </View>
    <View>
      <View>
        <View
          style={
            Object {
              "borderColor": "black",
              "borderStyle": "solid",
              "borderWidth": 2,
              "display": "flex",
              "flexDirection": "row",
              "justifyContent": "space-between",
            }
          }
        >
          <Text>
            my watchlist2
          </Text>
          <View
            accessible={true}
            isTVSelectable={true}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#007aff",
                "borderBottomWidth": null,
                "borderColor": "#007aff",
                "borderLeftWidth": null,
                "borderRadius": 5,
                "borderRightWidth": null,
                "borderTopWidth": null,
                "borderWidth": undefined,
                "elevation": 2,
                "flexDirection": "row",
                "height": 45,
                "justifyContent": "space-between",
                "opacity": 1,
                "paddingBottom": 6,
                "paddingTop": 6,
                "shadowColor": undefined,
                "shadowOffset": undefined,
                "shadowOpacity": undefined,
                "shadowRadius": undefined,
              }
            }
          >
            <Text />
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;
